### PR TITLE
Enhancement - Allows for inputting initial parameters programmatically…

### PR DESCRIPTION
**Pull request by travis-payne**
… via cmd line arguments.

I was attempting to run npx quorum-dev-quickstart via docker and realised it might be a little tricky to get through the prompt menu. To fix this problem I've used [https://www.npmjs.com/package/yargs](Yargs) to allow for either cmd line options or the original prompt. This doesn't affect the original method, however it does take an all or nothing approach (as in, either you give all the arguments via command line, or you use the prompt)

### CODE REVIEW

**Comment by joshuafernandes** [comment]
> … via cmd line arguments.
> 
> I was attempting to run npx quorum-dev-quickstart via docker and realised it might be a little tricky to get through the prompt menu. To fix this problem I've used [https://www.npmjs.com/package/yargs](Yargs) to allow for either cmd line options or the original prompt. This doesn't affect the original method, however it does take an all or nothing approach (as in, either you give all the arguments via command line, or you use the prompt)

Hi @travis-payne 

Thanks for the submission. Could you give us a little bit more context into this use case please? The tool especially in the npx flavour is intended to use used by and for developers to try out various options on the fly

Cheers

**Comment by travis-payne** [comment]
@joshuafernandes 
Hey, sure!

So I'm currently writing up some unit tests for some Besu functionality, and our current test ledger that we fire up is insufficient for what I'm writing, so I want to run this private network within a docker container. This is part of our automated test suite, so instead of simulating the input to the prompt that pops up when you run the npx command, I want to be able to just add some command line args, such like:
`npx quorum-dev-quickstart --elk true --privacy true --clientType besu `

To achieve the same result without having to step through the prompt.

**Comment by travis-payne** [comment]
@joshuafernandes Any update on this?

**Comment by joshuafernandes** [comment]
> @joshuafernandes Any update on this?

Aye, did some testing today on this and would like to request a couple of changes:
- set default options for all the args and make only the clientType and privacy as required please
- update the main readme with usage instructions as well for the new alternate flow

Also tried running the changes on a branch and it doesn't seem to pick up the option?

```
npm run build && npm run start --elk true --privacy true --clientType besu

> node build/index.js "besu"

Options:
  --help         Show help                                             [boolean]
  --version      Show version number                                   [boolean]
  --clientType   Ethereum client to use.
                               [string] [required] [choices: "besu", "goquorum"]
  --outputPath   Location for config files.             [string] [default: "./"]
  --elk          Enable support for logging with ELK.       [boolean] [required]
  --privacy      Enable support for private transactions    [boolean] [required]
  --orchestrate  Try out Codefi Orchestrate?                           [boolean]

Missing required arguments: clientType, elk, privacy
npm ERR! code ELIFECYCLE
npm ERR! errno 1

```

File: src/index.ts [hunk 1]
```diff
@@ -15,8 +16,30 @@ export async function main(): Promise<void> {
         process.exit(1);
     }
 
-    const qr = new QuestionRenderer(rootQuestion);
-    const answers = await qr.render();
+    let answers = {};
+
+    if(process.argv.slice(2).length > 0){
+      const args = await yargs(process.argv.slice(2)).options({
+        clientType: { type: 'string', demandOption: true, choices:['besu','goquorum'], describe: 'Ethereum client to use.' },
```
Comment (joshuafernandes) [inline src/index.ts:23]: Can we set the same defaults as the questions please for all options and make only the `clientType` and `privacy` as required?

**Review by joshuafernandes — CHANGES_REQUESTED** [review]
Hi @travis-payne, did some testing today on this and would like to request a couple of changes:
- set default options for all the args and make only the clientType and privacy as required please
- update the main readme with usage instructions as well for the new alternate flow

Also tried running the changes on a branch and it doesn't seem to pick up the options?
```
npm run build && npm run start --elk true --privacy true --clientType besu

> node build/index.js "besu"

Options:
  --help         Show help                                             [boolean]
  --version      Show version number                                   [boolean]
  --clientType   Ethereum client to use.
                               [string] [required] [choices: "besu", "goquorum"]
  --outputPath   Location for config files.             [string] [default: "./"]
  --elk          Enable support for logging with ELK.       [boolean] [required]
  --privacy      Enable support for private transactions    [boolean] [required]
  --orchestrate  Try out Codefi Orchestrate?                           [boolean]

Missing required arguments: clientType, elk, privacy
npm ERR! code ELIFECYCLE
npm ERR! errno 1
```

File: src/index.ts (hunk 1 above)
Comment (travis-payne) [reply src/index.ts:23]: @joshuafernandes  All done! 

As for your previous issue, you need to chuck an extra -- before you list the args:
``
npx quorum-dev-quickstart -- --clientType besu --outputPath ./ --elk true -privacy true --orchestra true
``

**Review by joshuafernandes — APPROVED** [review]

**Pull request merged**